src/action/tests/SummationActionTest.cpp

```diff
@@ -0,0 +1,200 @@
+#include "mesh/Edge.hpp"
+#include "mesh/Mesh.hpp"
+#include "mesh/Vertex.hpp"
+#include "mesh/config/DataConfiguration.hpp"
+#include "mesh/config/MeshConfiguration.hpp"
+#include "testing/Testing.hpp"
+
+using namespace precice;
+
+BOOST_AUTO_TEST_SUITE(ActionTests)
+BOOST_AUTO_TEST_SUITE(Summation, *testing::OnMaster())
+
+BOOST_AUTO_TEST_CASE(SummationTwoSources)
+{
+  using namespace mesh;
+  PtrMesh mesh(new Mesh("Mesh", 3, true, testing::nextMeshID()));
+  PtrData sourceData1   = mesh->createData("SourceData1", 1);
+  PtrData sourceData2	= mesh->createData("SourceData2", 1);
+  PtrData targetData   = mesh->createData("TargetData", 1);
+  std::vector<int> sourceDataID{sourceData1->getID(), sourceData2->getID()};	
+  int     		   targetDataID = targetData->getID();
+  mesh->createVertex(Eigen::Vector3d::Constant(0.0));
+  mesh->createVertex(Eigen::Vector3d::Constant(1.0));
+  mesh->createVertex(Eigen::Vector3d::Constant(2.0));
+
+  mesh->allocateDataValues();
+  auto &sourceValues1 = sourceData1->values();
+  auto &sourceValues2 = sourceData2->values();
+  auto &targetValues = targetData->values();
+  sourceValues1 << 2.0, 3.0, 4.0;
+  sourceValues2 << 4.0, 5.0, 1.0;
+  targetValues = Eigen::VectorXd::Zero(targetValues.size());
+
+  action::SummationAction sum(
+      action::SummationAction::ALWAYS_PRIOR, sourceDataID, targetDataID, mesh);
+
+  sum.performAction(0.0, 0.0, 0.0, 1.0);
+  BOOST_TEST(sourceValues1[0] == 2.0);
+  BOOST_TEST(sourceValues1[1] == 3.0);
+  BOOST_TEST(sourceValues1[2] == 4.0);
+  BOOST_TEST(sourceValues2[0] == 4.0);
+  BOOST_TEST(sourceValues2[1] == 5.0);
+  BOOST_TEST(sourceValues2[2] == 1.0);
+  BOOST_TEST(targetValues[0] == 6.0);
+  BOOST_TEST(targetValues[1] == 8.0);
+  BOOST_TEST(targetValues[2] == 5.0);
+
+  sum.performAction(0.0, 0.5, 0.5, 1.0);
+  BOOST_TEST(sourceValues1[0] == 2.0);
+  BOOST_TEST(sourceValues1[1] == 3.0);
+  BOOST_TEST(sourceValues1[2] == 4.0);
+  BOOST_TEST(sourceValues2[0] == 4.0);
+  BOOST_TEST(sourceValues2[1] == 5.0);
+  BOOST_TEST(sourceValues2[2] == 1.0);
+  BOOST_TEST(targetValues[0] == 6.0);
+  BOOST_TEST(targetValues[1] == 8.0);
+  BOOST_TEST(targetValues[2] == 5.0);
+
+  sum.performAction(0.0, 0.25, 0.75, 1.0);
+  BOOST_TEST(sourceValues1[0] == 2.0);
+  BOOST_TEST(sourceValues1[1] == 3.0);
+  BOOST_TEST(sourceValues1[2] == 4.0);
+  BOOST_TEST(sourceValues2[0] == 4.0);
+  BOOST_TEST(sourceValues2[1] == 5.0);
+  BOOST_TEST(sourceValues2[2] == 1.0);
+  BOOST_TEST(targetValues[0] == 6.0);
+  BOOST_TEST(targetValues[1] == 8.0);
+  BOOST_TEST(targetValues[2] == 5.0);
+
+  sum.performAction(0.0, 0.25, 1.0, 1.0);
+  BOOST_TEST(sourceValues1[0] == 2.0);
+  BOOST_TEST(sourceValues1[1] == 3.0);
+  BOOST_TEST(sourceValues1[2] == 4.0);
+  BOOST_TEST(sourceValues2[0] == 4.0);
+  BOOST_TEST(sourceValues2[1] == 5.0);
+  BOOST_TEST(sourceValues2[2] == 1.0);
+  BOOST_TEST(targetValues[0] == 6.0);
+  BOOST_TEST(targetValues[1] == 8.0);
+  BOOST_TEST(targetValues[2] == 5.0);
```

Comment:
I don't think we need all these cases with all different dt settings. More important would be to also have a test with vector-valued data, i.e. `dim=3`.